apps/server/src/modules/tool/controller/dto/response/tool-reference-list.response.ts

```diff
@@ -0,0 +1,11 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { ToolReferenceResponse } from './tool-reference.response';
+
+export class ToolReferenceListResponse {
```

Comment:
This is use allone, or in combinations with additional mappings to create pagination results. 
Make it really sense to have this response? 
If you return { data } on some places and on others { skip, data, total, ... } why not return always paginations and only have one mapper. Then this class can be deleted.

Reply:
Yes, it make sense and is a good practice to do it. Having a seperate object to encapsulate responses instead of returning an array give us the possibility to add things like pagination or additional metadata without have a breaking changes / backward incompatible changes.

Reply:
In addition to that. At the moment we dont have the requirement to implement pagination but i could imagine that this could come in the future :)